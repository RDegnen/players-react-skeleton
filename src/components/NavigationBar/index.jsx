import React from 'react';
import PropTypes from 'prop-types';
import { Navbar, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap';

const NavigationBar = ({ routes }) => (
  <Navbar>
    <NavbarBrand href="/">
      <h2>Players App</h2>
    </NavbarBrand>
    <Nav>
      {routes.map(({ path, name, nav }) => {
        if (!nav) return null;
        return (
          <NavItem>
            <NavLink href={path}>{name}</NavLink>
          </NavItem>
        );
      })}
    </Nav>
  </Navbar>
);

NavigationBar.propTypes = {
  routes: PropTypes.arrayOf(PropTypes.shape({
    path: PropTypes.string.isRequired,
    component: PropTypes.func,
    nav: PropTypes.bool,
    name: PropTypes.string.isRequired,
  })).isRequired,
};

export default NavigationBar;
