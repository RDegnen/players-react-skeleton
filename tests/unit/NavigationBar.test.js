import React from 'react';
import { shallow } from 'enzyme';
import { NavLink } from 'reactstrap';

import NavigationBar from '../../src/components/NavigationBar';

const defaultProps = {
  routes: [
    {
      path: '/register',
      component: jest.fn(),
      name: 'Register',
    },
    {
      path: '/roster',
      component: jest.fn(),
      nav: true,
      name: 'Roster',
    },
    {
      path: '/player/new',
      component: jest.fn(),
      nav: true,
      name: 'New Player',
    },
  ],
};

const getComponent = props => shallow(<NavigationBar {...defaultProps} {...props} />);

describe('<NavigationBar />', () => {
  it('should match snapshot', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('should have two NavLinks', () => {
    const component = getComponent();
    expect(component.find(NavLink).length).toBe(2);
  });
});
