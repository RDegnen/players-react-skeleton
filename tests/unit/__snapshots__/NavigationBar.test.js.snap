// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> should match snapshot 1`] = `
<Navbar
  expand={false}
  tag="nav"
>
  <NavbarBrand
    href="/"
    tag="a"
  >
    <h2>
      Players App
    </h2>
  </NavbarBrand>
  <Nav
    tag="ul"
    vertical={false}
  >
    <NavItem
      tag="li"
    >
      <NavLink
        href="/roster"
        tag="a"
      >
        Roster
      </NavLink>
    </NavItem>
    <NavItem
      tag="li"
    >
      <NavLink
        href="/player/new"
        tag="a"
      >
        New Player
      </NavLink>
    </NavItem>
  </Nav>
</Navbar>
`;
